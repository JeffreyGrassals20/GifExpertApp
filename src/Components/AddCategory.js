import React,{useState} from 'react'
import PropTypes from 'prop-types'


export const AddCategory = ({setCategories}) => {


    const [inputValue, setInput] = useState('')

    const handleInputChange = (e) => {
        setInput(e.target.value)

        console.log('Value has changed')
    } 

    const handleSubmit = (e) =>{
        e.preventDefault() 
        console.log('handleSubmit')
        if(inputValue.trim().length >= 1){
            setCategories(cats => [inputValue, ...cats])
            setInput('');
        }
    }
    return (
        <>
             <p> {inputValue}</p>
            <form onSubmit = {handleSubmit}>
            
                <h2>Add Category</h2>
                <input type="text" value = {inputValue} onChange = {handleInputChange}></input>
            </form>
        </>
    )
}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
