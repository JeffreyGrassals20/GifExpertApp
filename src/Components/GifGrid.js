import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import {GifGridItem} from './GifGridItem';

export const GifGrid = ({category}) => {

    const {data:images,loading} = useFetchGifs(category);
   
    return (
        <>
        <h2>{category} </h2>
        {loading && <h5 className= 'card animate__animated animate__flash'>Loading...</h5>}

        { <div className = 'card-grid'>
                <h4>
                    {
                        images.map((img) => (
                         <GifGridItem  {...img} key={img.id}/>
                        ))
                    }
                </h4>

        </div> }

        </>
    )
}
