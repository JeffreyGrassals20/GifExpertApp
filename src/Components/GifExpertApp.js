import React, {useState} from 'react';
import { AddCategory } from './AddCategory';
import { GifGrid } from './GifGrid';

const GifExpertApp  =() => {

    const  brands  = ['Basketball'];

    const [categories, setCategories] = useState(brands)

    /*const handleAdd = () => {
        setCategories([...categories, 'new one'])
    }*/

    return (
        <>
            <h2> Gif Expert </h2>
            <AddCategory setCategories = {setCategories}/>
            <hr />
            <ol>
                {
                    categories.map( category => 
                        <GifGrid key={category} category={category}/>
                   )
                }
            </ol>
        </>
    )
    
}
export default GifExpertApp
