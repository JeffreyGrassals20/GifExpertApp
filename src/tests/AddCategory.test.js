import React from 'react'
import{shallow}from 'enzyme'
import { AddCategory } from "../Components/AddCategory"


describe('AddCategory testing environment', () => {
    const setCategories = jest.fn();
    let wrapper = shallow(<AddCategory setCategories = {setCategories}/>)

    beforeEach(() =>{
        jest.clearAllMocks()
        wrapper = shallow(<AddCategory setCategories = {setCategories}/>)

    })

    test('should show correctly ', () => {
        expect(wrapper).toMatchSnapshot();
        
    })

    test('should change textBox', () => {
        const input = wrapper.find('input');
        const value = 'hello World'
        input.simulate('change', {target: {value}})

        expect(wrapper.find('p').text().trim()).toBe(value)
    })
    
    test('shouldnt return the submit ', () => {
        wrapper.find('form').simulate('submit', {preventDefault() {}})
        expect(setCategories).not.toHaveBeenCalled()
    })
    
    
})