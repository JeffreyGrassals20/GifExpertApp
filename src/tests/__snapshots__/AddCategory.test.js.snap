// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AddCategory testing environment should show correctly  1`] = `
<Fragment>
  <p>
     
  </p>
  <form
    onSubmit={[Function]}
  >
    <h2>
      Add Category
    </h2>
    <input
      onChange={[Function]}
      type="text"
      value=""
    />
  </form>
</Fragment>
`;
